designs/2019-config-tester/README.md

```diff
@@ -0,0 +1,145 @@
+<b>🔗PoC</b>: <a href="https://github.com/eslint/eslint/blob/2fb21b5dd52c81fe3c93cce0eb5fda3bf7789da0/lib/config-tester/config-tester.js#L60-L66">lib/config-tester/config-tester.js#L60-L66</a>
+</td></tr></table>
+
+### § `tester.run(targetName, options)` method
+
+Validates a config data.
+
+#### Parameters
+
+Name | Description
+:----|:-----------
+`targetName` | Required. If this package was a plugin, this is a config name of the plugin. Otherwise, this is a path to a file (relative from `projectRoot`).
+`options.ignoreDeprecatedRules` | Default is `false`. If `true` then the tester ignores deprecated rules.
+`options.ignoreDisabledUnknownRules` | Default is `false`. If `true` then the tester ignores unknown rules if the rule was configured as `0` (`"off"`).
+`options.ignoreRulesMissingFromConfig` | Default is `false`. If `true` then the tester ignores missing rules. The missing rules mean the rules that ESLint or a plugin defined but not configured.
+`options.ignoreMissingDependencies` | Default is `false`. If `true` then the tester ignores wrong dependency definition (`dependencies`/`peerDependencies`).
+
+#### Behavior
+
+Similarly to `RuleTester`, `ConfigTester` defines tests by `describe` and `it` global variables. Then it does:
+
+1. Validate the config object has the valid scheme with `validateConfigSchema()`.
+    <table><tr><td>
+    <b>🔗PoC</b>: <a href="https://github.com/eslint/eslint/blob/2fb21b5dd52c81fe3c93cce0eb5fda3bf7789da0/lib/config-tester/config-tester.js#L244-L248">lib/config-tester/config-tester.js#L244-L248</a>
+    </td></tr></table>
+1. Validate the config content with `validateConfigArray()`.
+    <table><tr><td>
+    <b>🔗PoC</b>: <a href="https://github.com/eslint/eslint/blob/2fb21b5dd52c81fe3c93cce0eb5fda3bf7789da0/lib/config-tester/config-tester.js#L250-L265">lib/config-tester/config-tester.js#L250-L265</a>
+    </td></tr></table>
+1. Report non-existence rules.
+    - Because `validateConfigArray(configArray)` ignores non-existence rules.
+    - Configured plugin's rules are in `configArray.pluginRules`.
+    - If `ignoreDisabledUnknownRules` option was `true` and non-existence rule's severity was `"off"`, the tester ignores it.
+    <table><tr><td>
+    <b>🔗PoC</b>: <a href="https://github.com/eslint/eslint/blob/2fb21b5dd52c81fe3c93cce0eb5fda3bf7789da0/lib/config-tester/config-tester.js#L267-L299">lib/config-tester/config-tester.js#L267-L299</a>
+    </td></tr></table>
+1. Report deprecated rules.
+    - If `ignoreDeprecatedRules` option was `true`, the tester skips this step.
+    - If the rule severity was `"off"`, the tester ignores it.
+    - Check `meta.deprecated` in both core rules and `configArray.pluginRules`.
+    <table><tr><td>
+    <b>🔗PoC</b>: <a href="https://github.com/eslint/eslint/blob/2fb21b5dd52c81fe3c93cce0eb5fda3bf7789da0/lib/config-tester/config-tester.js#L301-L338">lib/config-tester/config-tester.js#L301-L338</a>
+    </td></tr></table>
+1. Check whether the config congiures all rules.
```

Comment:
Typo: congiures --> configures

(Sorry this isn't a GitHub suggestion, it's not available on mobile phone)